### ISSUE TITLE
Should training a LoRa be this slow on RTX 3080?

### ISSUE THREAD

**Issue by DanteTheCreator** [bmaltais/kohya_ss#1837]
accelerate launch --num_cpu_threads_per_process=2 "./sdxl_train_network.py" --enable_bucket
                         --min_bucket_reso=256 --max_bucket_reso=2048
                         --pretrained_model_name_or_path="stabilityai/stable-diffusion-xl-base-1.0"
                         --train_data_dir="C:/Users/tatog/OneDrive/დოკუმენტები/AI/photos/img" --resolution="1024,1024"
                         --output_dir="C:/Users/tatog/OneDrive/დოკუმენტები/AI/photos/model" --network_alpha="1"
                         --save_model_as=safetensors --network_module=networks.lora --text_encoder_lr=5e-05
                         --unet_lr=0.0001 --network_dim=8 --gradient_accumulation_steps=8 --output_name="last"
                         --lr_scheduler_num_cycles="1" --no_half_vae --learning_rate="4e-07"
                         --lr_scheduler="constant_with_warmup" --lr_warmup_steps="213" --train_batch_size="1"
                         --max_train_steps="213" --save_every_n_epochs="1" --mixed_precision="fp16"
                         --save_precision="fp16" --cache_latents --optimizer_type="AdamW" --max_grad_norm="1"
                         --max_data_loader_n_workers="10" --bucket_reso_steps=64 --gradient_checkpointing --xformers
                         --persistent_data_loader_workers --bucket_no_upscale --noise_offset=0.0
                         
                       
                       
                       This is the command for now. I managed to get it working like this, but it is very slow. Takes minutes for a single step. What am I doing wrong?

**Comment by FurkanGozukara** [comment]
that means it is using shared VRAM

how much VRAM that GPU has?

**Comment by DanteTheCreator** [comment]
8 GB Dedicated and 16 GB Shared. So it is not enough?

**Comment by DanteTheCreator** [comment]
What is the average time for training LoRa?

**Comment by FurkanGozukara** [comment]
> 8 GB Dedicated and 16 GB Shared. So it is not enough?

yes with 8 gb of card you can only do sd 1.5 LoRA

**Comment by DanteTheCreator** [comment]
I am running 1.5, right now and it's still taking over 2 hours and is only on 50%

**Comment by FurkanGozukara** [comment]
> I am running 1.5, right now and it's still taking over 2 hours and is only on 50%

tell me it speed. it / second

**Comment by DanteTheCreator** [comment]
581/760 [2:55:17<54:00, 18.10s/it, avr_loss=0.359]

Currently, same operation.

**Comment by DanteTheCreator** [comment]
How do I tell it not to? Is there a setting or additional parameters?

**Comment by FurkanGozukara** [comment]
> How do I tell it not to? Is there a setting or additional parameters?

open task manager and look how much shared vram used

**Comment by DanteTheCreator** [comment]
I have been doing that. It uses all 24. 8 dedicated and 16 shared. How do I tell it to only use dedicated?

**Comment by FurkanGozukara** [comment]
> I have been doing that. It uses all 24. 8 dedicated and 16 shared. How do I tell it to only use dedicated?

you cnat tell you need to reduce vram demand

**Comment by DKnight54** [comment]
I see the main reason you are running out of Dedicated VRAM is probably the size of the image dataset (1024x1024) with batch size 1. You probably want to try another optimizer. AdamW uses roughly double of AdamW8bit, but AdamW8bit needs bitsandbytes installed. Looking at the requirements_windows_torch2.txt suggests it's installed, so you can try that.

Alternatively, another low VRAM optimizer is Adafactor. In theory, main benefit of Adafactor is low VRAM adaptive learning rate, but that can be turned off if you prefer constant learning rate.

Fixed rate learning parameters for Adafactor copied from the readme:

```
optimizer_type = "adafactor"
optimizer_args = [ "scale_parameter=False", "relative_step=False", "warmup_init=False" ]
lr_scheduler = "constant_with_warmup"
lr_warmup_steps = 100
learning_rate = 4e-7 # SDXL original learning rate
```

You can also read more about other optimizers **[here](https://rentry.org/59xed3#optimizers)**

Hope this helps you get the VRAM usage under control, and maybe even sneak in an increase to the batch size!

**Comment by joneschunghk** [comment]
I have tested some versions with the same dataset and configuration json file on my RTX-3060. Each test runs over 20 steps.

> accelerate launch --num_cpu_threads_per_process=4 "./sdxl_train_network.py" --enable_bucket --min_bucket_reso=256 --max_bucket_reso=2048 --pretrained_model_name_or_path="D:/AI/Stable Diffusion/sd.webui/webui/models/Stable-diffusion/something.safetensors" --train_data_dir="D:/AI/Stable Diffusion/Datasets/V3/Test\img" --resolution="1024,1024" --output_dir="D:/AI/Stable Diffusion/Datasets/V3/Test\model" --logging_dir="D:/AI/Stable Diffusion/Datasets/V3/Test\log" --network_alpha="1" --save_model_as=safetensors --network_module=networks.lora --text_encoder_lr=0.0002 --unet_lr=0.0004 --network_dim=16 --output_name="Lora_XL10_test" --lr_scheduler_num_cycles="10" --cache_text_encoder_outputs --no_half_vae --full_bf16 --learning_rate="0.0004" --lr_scheduler="cosine" --lr_warmup_steps="60" --train_batch_size="2" --max_train_steps="6000" --save_every_n_epochs="1" --mixed_precision="bf16" --save_precision="bf16" --caption_extension=".txt" --cache_latents --cache_latents_to_disk --optimizer_type="Adafactor" --optimizer_args scale_parameter=False relative_step=False warmup_init=False --max_data_loader_n_workers="0" --bucket_reso_steps=64 --gradient_checkpointing --xformers --bucket_no_upscale --noise_offset=0.0357 --network_train_unet_only --sample_sampler=euler_a --sample_prompts="D:/AI/Stable Diffusion/Datasets/V3/Test\model\sample\prompt.txt" --sample_every_n_epochs="1"

v22.0.0 (3b83a1c) - 3.46s/it
v22.1.0 (0ce88bb) - 3.11s/it
v22.1.1 (9eb5da2) - 3.30s/it
v22.2.0 (6adfb3f) - 28.91s/it
v22.4.1 (842d9c7) - 26.79s/it

It seems to be some problems between v22.1.1 to v22.2.0

**Comment by nokeyfan** [comment]
i still use version 22.0.1 which gives me about 2.50 s/it. The newer version from 22.2 and higher gives about 48 s/it on a rtx 2070 8GB.  Similar results like (joneschunghk) posted above.

**Comment by FurkanGozukara** [comment]
> i still use version 22.0.1 which gives me about 2.50 s/it. The newer version from 22.2 and higher gives about 48 s/it on a rtx 2070 8GB. Similar results like (joneschunghk) posted above.

that is happening because of using shared vram. something causes more vram usage

**Comment by nokeyfan** [comment]
Just for the info, the new version of today 22.6.0, the speed is fast again. On rtx2070 8GB, 1.90 s/it. Thanks for the fix
